fdbserver/DataDistribution.actor.cpp

```diff
@@ -1753,9 +1760,15 @@ ACTOR Future<Void> auditStorageCore(Reference<DataDistributor> self,
 
 	try {
 		ASSERT(audit != nullptr);
+		if (audit->getContext() == DDAuditContext::ResumeAudit) {
```

Comment:
`resumeAudit()` is called when DD starts, why can we assert the new DDID is the same as the one generated by a previous DD?

Reply:
Also, where does DD update the `ddAuditId` when a new DD starts?

Reply:
When DD restarts, it calls runAuditStorage to resume an audit. When runAuditStorage, auditState.ddAuditId is set as the new DD id.

Reply:
Following important invariant is obeyed. Any ongoing DDAudit is triggered by runAuditStorage and runAuditStorage always sets the current dd as the audit->auditState.ddAuditId. As a result, any ongoing DDAudit->auditState.ddAuditId is set as the current dd id. Since the persist auditState can be only updated by an ongoing DDAudit of the latest dd (using dd lock). So, auditState always keeps the latest dd id.